### PR TITLE
5% block reward code to AVN foundation wallet

### PR DESCRIPTION
**Pull request by alamshafil**
5% block reward code to AVN foundation wallet

### CODE REVIEW

**Comment by alamshafil** [comment]
Important to have #71 to scan code. Rebasing against updated 4.1 branch.

File: src/chainparams.cpp [hunk 1]
```diff
@@ -215,6 +215,10 @@ class CMainParams : public CChainParams {
         // Avian BIP44 cointype in mainnet is '175'
         nExtCoinType = 175;
 
+        vector<FounderRewardStructure> rewardStructures = {  {INT_MAX, 5}// 5% founder/dev fee forever
+                                        										   };
+        consensus.nFounderPayment = FounderPayment(rewardStructures, 250);
```
Comment (alamshafil) [inline src/chainparams.cpp:220]: "250" needs to be updated to the latest block height

**Comment by alamshafil** [comment]
Since this PR deals with a consensus change, I believe having #72 is important to check that nothing broke or regressed with this PR.

File: src/blockencodings.cpp [hunk 2]
```diff
@@ -199,6 +199,7 @@ ReadStatus PartiallyDownloadedBlock::FillBlock(CBlock& block, const std::vector<
         return READ_STATUS_INVALID;
 
     CValidationState state;
+        //skip founder check
```
[the remark this answers is not in the document]
Comment (decentralisedcoder) [reply src/blockencodings.cpp]: This is legacy and can be removed

File: src/chainparams.cpp [hunk 3]
```diff
@@ -215,6 +215,10 @@ class CMainParams : public CChainParams {
...
+
```
[the remark this answers is not in the document]
Comment (decentralisedcoder) [reply src/chainparams.cpp]: Yes nice easy change

File: src/chainparams.cpp [hunk 4]
```diff
@@ -565,6 +578,10 @@ class CRegTestParams : public CChainParams {
         fMineBlocksOnDemand = true;
         fMiningRequiresPeers = false;
 
+        vector<FounderRewardStructure> rewardStructures = {  {INT_MAX, 5}// 5% founder/dev fee forever
+                                                        										   };
+		consensus.nFounderPayment = FounderPayment(rewardStructures, 200);
+
```
[the remark this answers is not in the document]
Comment (decentralisedcoder) [reply src/chainparams.cpp]: Yes there should be this for testnet/regtest and mainnet is explicitly defined

File: src/consensus/tx_verify.cpp [hunk 5]
```diff
@@ -376,17 +376,20 @@ bool CheckTransaction(const CTransaction& tx, CValidationState &state, bool fChe
 
     if (tx.IsCoinBase())
     {
-        if (tx.vin[0].scriptSig.size() < 2 || tx.vin[0].scriptSig.size() > 100)
-            return state.DoS(100, false, REJECT_INVALID, "bad-cb-length");
-
-        if (AreCoinbaseCheckAssetsDeployed()) {
```
[the remark this answers is not in the document]
Comment (decentralisedcoder) [reply src/consensus/tx_verify.cpp]: yes once we have 4.1 out of the way this will help with 4.2 asset release

File: src/chainparams.cpp (hunk 1 above)
Comment (decentralisedcoder) [reply src/chainparams.cpp]: Agreed as we will be activating down the line to avoid problems with existing chain work

**Comment by alamshafil** [comment]
Working as expected on regtest.

`FounderPayment::FillFounderPayment -- Founder payment 3125000000 to n1BurnXXXXXXXXXXXXXXXXXXXXXXU1qejP`

```
   {
      "value": 31.25000000,
      "n": 1,
      "scriptPubKey": {
        "asm": "OP_DUP OP_HASH160 d7c8944771bbfe427418f27320e72a1322faf134 OP_EQUALVERIFY OP_CHECKSIG",
        "hex": "76a914d7c8944771bbfe427418f27320e72a1322faf13488ac",
        "reqSigs": 1,
        "type": "pubkeyhash",
        "addresses": [
          "n1BurnXXXXXXXXXXXXXXXXXXXXXXU1qejP"
        ]
      }
    }
 ```

**Comment by alamshafil** [comment]
Todo:

- [ ] Get block height
- [ ] Fix dev address

**Pull request merged**